File: src/main/java/ninja/hudy/infosmog/loader/ApplicationLoader.java
package ninja.hudy.infosmog.loader;

import ninja.hudy.infosmog.scheduler.ReaderScheduler;
import ninja.hudy.infosmog.update.Configurator;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

@Component
public class ApplicationLoader implements CommandLineRunner {
    @Autowired
    private Configurator configurator;

    @Autowired
    private ReaderScheduler readerScheduler;

    @Override
    public void run(String... strings) throws Exception {
        configurator.configure();
        readerScheduler.read();
        configurator.assignKnownStations();
    }


}

File: src/main/kotlin/pl/org/smok/infosmog/loader/ApplicationLoader.kt
package pl.org.smok.infosmog.loader

import pl.org.smok.infosmog.scheduler.ReaderScheduler
import pl.org.smok.infosmog.update.Configurator
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class ApplicationLoader : CommandLineRunner {
    @Autowired
    lateinit var configurator: Configurator;

    @Autowired
    lateinit var readerScheduler: ReaderScheduler;

    @Throws(Exception::class)
    override fun run(vararg strings: String) {
        configurator.configure()
        readerScheduler.read()
        configurator.assignKnownStations()
    }


}
